### PR TITLE
added SchemaRegistryConfigBasicAuthCredentialsSource

### PR DESCRIPTION
**Pull request by mhowlett**
addresses #679. this was accidentally omitted from 1.0-beta2, probably because at one point i was considering not including it.

### CODE REVIEW

File: src/Confluent.SchemaRegistry/SchemaRegistryConfig.cs [hunk 1]
```diff
@@ -53,6 +53,8 @@ public static class PropertyNames
 
             /// <summary>
             ///     Specifies the configuration property(ies) that provide the basic authentication credentials.
+            ///     USER_INFO: credentials are specified via the `schema.registry.basic.auth.user.info` config property in the form username:password.
+            ///     SASL_INHERIT: credentials are specified via the `sasl.username` and `sasl.password` configuration properties.
```
Comment (edenhill) [inline src/Confluent.SchemaRegistry/SchemaRegistryConfig.cs]: Shouldn't this be an enum thing?

Comment (mhowlett) [reply src/Confluent.SchemaRegistry/SchemaRegistryConfig.cs]: it shouldn't be here - this string values would be best to match the JAVA config values. But it should be in the static config class, following the pattern of other configs, and the property should translate between the two names.

good catch.

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs [hunk 2]
```diff
@@ -81,29 +81,22 @@ public CachedSchemaRegistryClient(IEnumerable<KeyValuePair<string, string>> conf
...
-                if (basicAuthInfo == "")
-                {
-                    throw new ArgumentException($"CachedSchemaRegistryClient: {SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthCredentialsSource} set to 'USER_INFO', but no value specified for {SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthUserInfo}.");
-                }
-                var userPass = (basicAuthInfo).Split(':');
-                if (userPass.Length != 2)
+                if (basicAuthInfo != "")
                 {
-                    throw new ArgumentException($"CachedSchemaRegistryClient: Configuration property {SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthUserInfo} must be of the form 'username:password'.");
+                    var userPass = (basicAuthInfo).Split(':');
```
Comment (edenhill) [inline src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs]: We should support escaping the :

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs [hunk 3]
```diff
@@ -126,7 +119,7 @@ public CachedSchemaRegistryClient(IEnumerable<KeyValuePair<string, string>> conf
             }
             else
             {
-                // no basic auth info.
+                throw new ArgumentException($"CachedSchemaRegistryClient: Invalid value '{basicAuthSource}' specified for property '{SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthCredentialsSource}'");
```
Comment (edenhill) [inline src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs]: nit: hint of the correct values.

File: src/Confluent.SchemaRegistry/SchemaRegistryConfig.cs [hunk 4]
```diff
@@ -14,12 +14,30 @@
...
+        ///     Credentials are specified via the `schema.registry.basic.auth.user.info` config property in the form username:password.
+        ///     If `schema.registry.basic.auth.user.info` is not set, authentication is disabled.
+        /// </summary>
+        UserInfo,
+
+        /// <summary>
+        ///     Credentials are specified via the `sasl.username` and `sasl.password` configuration properties.
+        /// </summary>
+        SaslInherit
+    }
```
Comment (edenhill) [inline src/Confluent.SchemaRegistry/SchemaRegistryConfig.cs]: what about retrieving it from the URL?

Comment (mhowlett) [reply src/Confluent.SchemaRegistry/SchemaRegistryConfig.cs]: left as a todo. I had problems getting this to work and just omitted it in the end.

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs (hunk 2 above)
Comment (milindl) [reply src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs:92]: For the sake of documenting this - the RFC for basic auth disallows any colon in the username, and rather everything after the first colon should be considered to be a part of the password.

See #1767

**Pull request merged**